src/modals/CollectArticleModal.tsx

```diff
@@ -0,0 +1,226 @@
+import { t, Trans } from '@lingui/macro';
+import { EVMChainResolver } from '@masknet/web3-providers';
+import { useQuery } from '@tanstack/react-query';
+import { estimateFeesPerGas, getBalance, switchChain } from '@wagmi/core';
+import { forwardRef, useMemo, useState } from 'react';
+import { useAsync, useAsyncFn } from 'react-use';
+import { useAccount, useChains } from 'wagmi';
+
+import LoadingIcon from '@/assets/loading.svg';
+import { Avatar } from '@/components/Avatar.js';
+import { ClickableButton } from '@/components/ClickableButton.js';
+import { CloseButton } from '@/components/CloseButton.js';
+import { Modal } from '@/components/Modal.js';
+import { config } from '@/configs/wagmiClient.js';
+import { enqueueErrorMessage, enqueueSuccessMessage } from '@/helpers/enqueueMessage.js';
+import { formatEthereumAddress } from '@/helpers/formatEthereumAddress.js';
+import { getSnackbarMessageFromError } from '@/helpers/getSnackbarMessageFromError.js';
+import { multipliedBy, rightShift, ZERO } from '@/helpers/number.js';
+import { resolveArticleCollectProvider } from '@/helpers/resolveArticleCollectProvider.js';
+import { useSingletonModal } from '@/hooks/useSingletonModal.js';
+import { type SingletonModalRefCreator } from '@/libs/SingletonModal.js';
+import { ConnectWalletModalRef } from '@/modals/controls.js';
+import { type Article, ArticlePlatform } from '@/providers/types/Article.js';
+
+export interface CollectArticleModalOpenProps {
+    article: Article;
+}
+
+export const CollectArticleModal = forwardRef<SingletonModalRefCreator<CollectArticleModalOpenProps>>(
+    function CollectArticleModal(_, ref) {
+        const account = useAccount();
+        const chains = useChains();
+
+        const [props, setProps] = useState<CollectArticleModalOpenProps>();
+        const [open, dispatch] = useSingletonModal(ref, {
+            onOpen: (props) => {
+                setProps({
+                    article: props.article,
+                });
+            },
+            onClose: () => {},
+        });
+
+        const { data, isLoading: queryDetailLoading } = useQuery({
+            enabled: !!props?.article && open,
+            queryKey: ['article', props?.article.platform, props?.article.id, props?.article.origin],
+            queryFn: async () => {
+                if (!props?.article?.origin) return;
+                const provider = resolveArticleCollectProvider(props.article.platform);
+
+                return provider.getArticleDetail(
+                    props.article.platform === ArticlePlatform.Paragraph ? props.article.origin : props.article.id,
```

Comment:
```suggestion
                return provider.getArticleDetail(resolveArticleId(props.article),
```

**resolveArticleId.ts**

```ts
export function getArticleId(article: Article) {
    switch (article.platform) {
        // enumerate all the platforms
    }
}
```